File: src/main/java/me/ag2s/epublib/util/NoCloseWriter.java
package me.ag2s.epublib.util;

import java.io.IOException;
import java.io.Writer;

public class NoCloseWriter extends Writer {
   private final Writer writer;

   public NoCloseWriter(Writer writer) {
      this.writer = writer;
   }

   public void close() {
   }

   public void flush() throws IOException {
      this.writer.flush();
   }

   public void write(char[] cbuf, int off, int len) throws IOException {
      this.writer.write(cbuf, off, len);
   }
}

File: src/main/kotlin/me/ag2s/epublib/util/NoCloseWriter.kt
package me.ag2s.epublib.util

import java.io.Writer

class NoCloseWriter(private val writer: Writer) : Writer() {
    override fun close() = Unit
    override fun flush() = writer.flush()
    override fun write(cbuf: CharArray, off: Int, len: Int) = writer.write(cbuf, off, len)
}
